documentation_quality_analysis/analyze_library/doc_parser/doc_signature_util.py

```diff
@@ -43,27 +41,46 @@ def get_signatures_from_section(section) -> List[Signature]:
         description = "".join(desc).strip()
 
         try:
-            if "class" in description:
-                class_element = re.findall(re.compile(r'class\s((?:.*)(?:\(.*\))?)'), description)
-                if len(class_element) > 0:
-                    class_name = re.findall('(.*)\(', class_element[0])[0] if "(" in class_element[0] else class_element[0].strip()
-                    class_signature: ClassConstructorSignature = _get_parsed_class_details(class_element[0])
-                    signatures.append(class_signature)
-            elif "(" in description:
-                code = re.findall('[A-Za-z]*\s(.+\(.*\))', description)
-                if len(code) > 0:
-                    method = _get_parsed_method_details(method=code[0], class_name=class_name)
-                    signatures.append(method)
-                else:
-                    method = _get_parsed_method_details(method=description, class_name=class_name)
-                    signatures.append(method)
-
```

Comment:
does it makes sense to somehow keep this "more generic" parsing mechanism still available in the code? For example, have different parsing strategies that can be called/used, and then just use the specific one for the format?